src/requests/payrequest.cpp

```diff
@@ -10,10 +10,11 @@ Router::Status payWorkerHandler(const Router::vars_t& vars, const graft::Input&
     PayRequest in = input.get<PayRequest>();
     if (!in.Address.empty() && !in.Amount.empty()
             && !in.PaymentID.empty() && in.BlockNumber >= 0 //TODO: Change to `BlockNumber > 0`
-            && ctx.global.hasKey(in.PaymentID + CONTEXT_KEY_SALE_STATUS))
+            && ctx.global.hasKey(in.PaymentID + CONTEXT_KEY_STATUS))
```

Comment:
Current implementation of _global context_ gives no guarantee an element was not deleted after `hasKey()` check. In cases like that, we can retrieve a value first, then check the value vs. default one. E.g.
```
int current_status = ctx.global[in.PaymentID + CONTEXT_KEY_SALE_STATUS];
if (current_status != 0)
{
    ...
}
```

If you see this as an issue, let's discuss it